### DISCUSSION TITLE
Items generated by addLevel are not unique?

### DISCUSSION THREAD

**Discussion by joshuacurtiss** [replit/kaboom#161] [General]
Hey everyone. I thought I'd just bounce this off you before making an issue. I've been trying to generate interactive items (enemies, coins) on maps using the `addLevel()` functionality that enables you to generate items based on the characters on your map. However, when I do that, animations and other actions seem to be linked to all the items instead of individual to each generated item. Is this by design? I just figured out that I can get around it by using my own routine for finding and `add()`ing items explicitly. Thoughts? 

Thanks, I'm loving the framework. Using the standard 0.5.1 on npm, by the way.

Josh

**Comment by slmjkdbtl** [comment]
If it involves dynamic content, can pass a function that returns the comp array. It'll be evaluated on every instance.

```js
"#" : () => [
    comp(rand()),
    "tag",
],
```

**Reply by joshuacurtiss** [reply]
Gotcha, thanks!
